src/test/java/com/kdt/jpaproject/domain/PersistenceContextTest.java

```diff
@@ -0,0 +1,193 @@
+package com.kdt.jpaproject.domain;
+
+import lombok.extern.slf4j.Slf4j;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+
+import javax.persistence.EntityManager;
+import javax.persistence.EntityManagerFactory;
+import javax.persistence.EntityTransaction;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+@Slf4j
+@SpringBootTest
+class PersistenceContextTest {
+
+    @Autowired
+    CustomerRepository repository;
+
+    @Autowired
+    EntityManagerFactory entityManagerFactory;
+
+    @BeforeEach
+    void setUp() {
+        repository.deleteAll();
+    }
+
+    @Test
+    void 저장() {
+        EntityManager entityManager = entityManagerFactory.createEntityManager();
+        EntityTransaction transaction = entityManager.getTransaction();
+
+        transaction.begin();
+
+        Customer customer = new Customer();  // 비영속상태
+        customer.setId(1L);
+        customer.setFirstName("eunbi");
+        customer.setLastName("choi");
+
+        entityManager.persist(customer);     // 비영속 -> 영속 (영속화)
+        transaction.commit();               // entityManager.flush();
+    }
+
+    @Test
+    void 조회_DB조회_detach() {
+        EntityManager entityManager = entityManagerFactory.createEntityManager();
+        EntityTransaction transaction = entityManager.getTransaction();
+
+        transaction.begin();
+        Customer customer = new Customer();
+        customer.setId(1L);
+        customer.setFirstName("eunbi");
+        customer.setLastName("choi");
+
+        entityManager.persist(customer);
+        transaction.commit();
+
+        entityManager.detach(customer);     // 영속 -> 준영속
+
+        Customer selectedCustomer = entityManager.find(Customer.class, 1L);     // select
+        log.info("{} {}", selectedCustomer.getFirstName(), selectedCustomer.getFirstName());
+
+        assertThat(selectedCustomer)
+                .hasFieldOrPropertyWithValue("id", customer.getId())
+                .hasFieldOrPropertyWithValue("firstName", customer.getFirstName())
+                .hasFieldOrPropertyWithValue("lastName", customer.getLastName());
+    }
+
+    @Test
+    void 조회_DB조회_clear() {
+        EntityManager entityManager = entityManagerFactory.createEntityManager();
+        EntityTransaction transaction = entityManager.getTransaction();
```

Comment:
```suggestion
private EntityManager entityManager;
	private EntityTransaction transaction;
	
	@BeforeEach
	void setUp() {
		entityManager = entityManagerFactory.createEntityManager();
		transaction = entityManager.getTransaction();
	}
```

공통적인 내용은 외부로 빼도 좋을 것 같습니다.!